lib/std/fs.zig

```diff
@@ -1655,6 +1664,48 @@ pub fn selfExePath(out_buffer: *[MAX_PATH_BYTES]u8) SelfExePathError![]u8 {
             // TODO could this slice from 0 to out_len instead?
             return mem.spanZ(@ptrCast([*:0]u8, out_buffer));
         },
+        .openbsd => {
+            // OpenBSD has no support for this. Out of the various tried methods,
+            // the only working workaround comes from stackoverflow.com/a/31495527
+
+            // Retrieve the argument vector
+            var mib = [4]c_int{ os.CTL_KERN, os.KERN_PROC_ARGS, os.system.getpid(), os.KERN_PROC_ARGV };
+            var argv: [os.ARG_MAX / @sizeOf([]u8)][*:0]u8 = undefined;
+            var len: usize = undefined;
+            try os.sysctl(&mib, &argv, &len, null, 0);
+
+            var prg_name = argv[0];
+            if (prg_name[0] == '/' or prg_name[0] == '.') {
+                return os.realpathZ(prg_name, out_buffer);
+            }
+            prg_name = mem.spanZ(prg_name);
```

Comment:
This code assumes pathname beginning with `.` is a path component of either `.` or `..` but it could also be a dot-named file. Let's use `std.mem.indexOfScalar(..., '/')` to detect `idx == 0` or `"."` or `".."` here.